src/api/BoreholeLockService.cs

```diff
@@ -42,28 +42,48 @@ public async Task<bool> IsUserLackingPermissions(int? boreholeId, string? subjec
         return IsUserLackingPermissions(borehole, user);
     }
 
-    /// <inheritdoc />
-    public bool IsUserLackingPermissions(ICollection<Borehole> boreholes, User user)
-    {
-        return boreholes.Any(borehole => IsUserLackingPermissions(borehole, user));
-    }
-
     private bool IsUserLackingPermissions(Borehole borehole, User user)
     {
-        if (borehole.Workflows != null)
+        try
         {
-            var boreholeWorkflowRoles = borehole.Workflows
-                .Select(w => w.Role)
-                .ToHashSet();
+            if (borehole.Workflows == null || borehole.Workflows.Count == 0)
+            {
+                logger.LogWarning("User with SubjectId <{SubjectId}> attempted to edit BoreholeId <{BoreholeId}>, but it has no workflows.", user.SubjectId, borehole.Id);
+                return true;
+            }
+
+            var boreholeWorkflowRoles = borehole.Workflows.Select(w => w.Role).ToHashSet();
 
             if (user.WorkgroupRoles == null || !user.WorkgroupRoles.Any(x => x.WorkgroupId == borehole.WorkgroupId && boreholeWorkflowRoles.Contains(x.Role)))
             {
-                logger.LogWarning("Current user with subject_id <{SubjectId}> does not have the required role to edit the borehole with id <{BoreholeId}>.", user.SubjectId, borehole.Id);
+                logger.LogWarning("User with SubjectId <{SubjectId}> lacks the required role to edit BoreholeId <{BoreholeId}>.", user.SubjectId, borehole.Id);
                 return true;
             }
+
+            return false;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Error checking user permissions for BoreholeId <{BoreholeId}>.", borehole.Id);
+            return true;
         }
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> HasUserWorkgroupPermissionsAsync(int? boreholeId, string? subjectId)
+    {
+        var user = await GetUserWithWorkgroupRolesAsync(subjectId).ConfigureAwait(false);
+
+        if (user.IsAdmin) return false;
```

Comment:
REQ: Ich denke das ist falsch; so wie ich es interpretiere, hat dann kein Admin Rechte in den Workgroups?